metis-enrichment/metis-enrichment-common/src/main/java/eu/europeana/enrichment/api/internal/FieldType.java

```diff
@@ -1,213 +1,48 @@
-
-  DCTERMS_SPATIAL(Choice::ifSpatial, Choice::getSpatial, Choice::setSpatial, Spatial::new,
-      EntityType.PLACE),
-
-  DC_SUBJECT(Choice::ifSubject, Choice::getSubject, Choice::setSubject, Subject::new,
-      EntityType.CONCEPT),
-
-  DCTERMS_MEDIUM(Choice::ifMedium, Choice::getMedium, Choice::setMedium, Medium::new,
-      EntityType.CONCEPT),
-
-  DC_FORMAT(Choice::ifFormat, Choice::getFormat, Choice::setFormat, Format::new,
-      EntityType.CONCEPT);
-
-  private final ChoiceContentHandler<?> choiceContentHandler;
-  private final EntityType entityType;
-
-  <T extends ResourceOrLiteralType> FieldType(Predicate<Choice> choiceChecker,
-      Function<Choice, T> contentGetter, BiConsumer<Choice, T> contentSetter,
-      Supplier<T> contentCreator, EntityType entityType) {
-    this.choiceContentHandler =
-        new ChoiceContentHandler<>(choiceChecker, contentGetter, contentSetter, contentCreator);
-    this.entityType = entityType;
-  }
+public interface FieldType {
 
   /**
-   * Extract fields from a Proxy for enrichment
-   *
-   * @param proxy The proxy to use for enrichment
-   * @return A list of values (text and language) ready for enrichment
+   * @return the entity type associated to this field - it is not null.
    */
-  public final Set<FieldValue> extractFieldValuesForEnrichment(ProxyType proxy) {
-    return extractFields(proxy)
-        .filter(content -> StringUtils.isNotEmpty(content.getString()))
-        .map(this::convert)
-        .collect(Collectors.toSet());
-  }
+  EntityType getEntityType();
 
   /**
-   * Extract resources from a Proxy for enrichment
+   * Extract the field values set from the provided about type.
+   * <p>It gets the values for the specific field and creates a set of {@link FieldValue}s</p>
    *
-   * @param proxy The proxy to use for enrichment
-   * @return A list of values ready for enrichment
+   * @param aboutType the about type to use
+   * @return the set of field values
    */
-  public final Set<String> extractFieldLinksForEnrichment(ProxyType proxy) {
-    return extractFields(proxy)
-        .map(ResourceOrLiteralType::getResource)
-        .filter(Objects::nonNull)
-        .map(Resource::getResource)
-        .filter(StringUtils::isNotBlank)
-        .collect(Collectors.toSet());
-  }
-
-  private Stream<? extends ResourceOrLiteralType> extractFields(ProxyType proxy) {
-    return Optional.ofNullable(proxy.getChoiceList()).stream().flatMap(Collection::stream)
-        .filter(choiceContentHandler.choiceChecker)
-        .map(choiceContentHandler.contentGetter)
-        .filter(Objects::nonNull);
-  }
-
-  private FieldValue convert(ResourceOrLiteralType content) {
-    final String language = content.getLang() == null ? null : content.getLang().getLang();
-    return new FieldValue(content.getString(), language);
+  default <T extends AboutType> Set<FieldValue> extractFieldValuesForEnrichment(T aboutType) {
```

Comment:
Just providing generic types in the methods like this does not add anything, I think (it's the same as just having a parameter of type `AboutType`). What I meant was declare the interface as
`public interface FieldType<T extends AboutType>`. 
And then using the T parameter as you are already doing in the methods.

Reply:
Indeed the first change doesn't add much, so I was wondering why you asked for that..
Anyway the change you mention here, although I'm not really a fan of that, I've made it.. 
So let me know if it's what you have in mind